server/src/controllers/blogController.js

```diff
@@ -0,0 +1,124 @@
+
+  static async createBlog(req, res) {
+    const img = await uploadToCloud(req.file);
+    if (!img) return Response.error(res, 500, "Failed to upload photo");
+    try {
+      const blog = await Blog.create({
+        title: req.body.title,
+        content: req.body.content,
+        blogImage: img.url,
+      });
+      Response.success(res, 201, "Sucessfully Created the blog", blog);
+    } catch (err) {
+      if (err.code === 11000)
+        return Response.error(res, 400, "Duplicate field value entered");
+      Response.error(res, 500, err);
+    }
+  }
+
+  static async getBlog(req, res) {
+    try {
+      const blog = await Blog.findOne({ _id: req.params.id }).populate(
+        "comments"
+      );
+      if (blog === null) return Response.error(res, 404, "Blog not found");
+      return Response.success(res, 200, "Sucessfully Retrieved the blog", blog);
+    } catch (err) {
+      Response.error(res, 404, "Blog not found");
+    }
+  }
+
+  static async updateBlog(req, res) {
+    const img = await uploadToCloud(req.file);
+    if (!img) return Response.error(res, 500, "Failed to upload photo");
+    try {
+      const blogUpdated = await Blog.findOneAndUpdate(
+        req.params.id,
+        {
+          title: req.body.title,
+          content: req.body.content,
+          blogImage: img.url,
+        },
+        { new: true }
+      );
+      if (blogUpdated === null)
+        return Response.error(res, 404, "Blog not found");
+      return Response.success(
+        res,
+        200,
+        "Sucessfully Updated the blog",
+        blogUpdated
+      );
+    } catch (error) {
+      return Response.error(res, 404, "Blog not found");
+    }
+  }
+
+  static async deleteBlog(req, res) {
+    try {
+      const deleteBlog = await Blog.findByIdAndRemove(req.params.id);
+      if (deleteBlog === null)
+        return Response.error(res, 404, "Blog not found");
+      return Response.success(
+        res,
+        200,
+        "Sucessfully deleted the blog",
+        deleteBlog
+      );
+    } catch (error) {
+      return Response.error(res, 404, "Blog not found");
+    }
+  }
+
+  static async addComment(req, res) {
+    const blogId = req.params.id;
+    if (!blogId) return Response.error(res, 400, "Please provide the blogId");
+    const blog = await Blog.findById(blogId);
+    if (!blog) return Response.success(res, 404, "Blog Not Found");
+    try {
+      const newComment = new Comment({ comment: req.body.comment });
+      await newComment.save();
+      blog.comments.push(newComment);
+      blog.commentCounts += 1;
+      await blog.save();
+      return Response.success(
+        res,
+        201,
+        "Comment sucessfully Created",
+        newComment
+      );
+    } catch (error) {
+      return Response.error(res, 500, "something went wrong");
+    }
+  }
+
+  static async addLike(req, res) {
+    const blogId = req.params.id;
+    if (!blogId) return Response.error(res, 400, "Please provide the blogId");
```

Comment:
Same here, you see that would be great if validate this param with Joi so that can be reused.